File: ProblemSet/53.maximum-subarray/java/53.maximum-subarray 贪心法.java
/*
 * @lc app=leetcode id=53 lang=java
 *
 * [53] Maximum Subarray
 */

// @lc code=start
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = Integer.MIN_VALUE, sum = 0;
        for (int num : nums) {
            sum += num;
            maxSum = Math.max(maxSum, sum);
            sum = Math.max(sum, 0);
        }
        return maxSum;
    }
}
// @lc code=end

File: ProblemSet/53.maximum-subarray/python/53.maximum-subarray 贪心法.py
import sys

#
# @lc app=leetcode id=53 lang=python3
#
# [53] Maximum Subarray
#

# @lc code=start
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum, cur = -sys.maxsize - 1, 0
        for num in nums:
            cur += num
            maxSum = max(maxSum, cur)
            cur = max(cur, 0) 
        return maxSum
# @lc code=end

